pkg/syncthing/client.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 The envd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syncthing
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"time"
+)
+
+func NewApiClient() *http.Client {
+	return &http.Client{
+		Timeout: 60 * time.Second,
+	}
+}
+
+const (
+	GET  = "GET"
+	POST = "POST"
+	PUT  = "PUT"
+)
+
+// Makes API calls to the syncthing instance's rest api
+func (s *Syncthing) ApiCall(method string, url string, params map[string]string, body []byte) ([]byte, error) {
```

Comment:
This abstraction is not suitable. You can refer to the implementation in https://github.com/tensorchord/envd-server/blob/main/client/client.go#L25

Reply:
Ok, let me take a closer look